build/pipelines/templates/run-staticAnalysis.yaml

```diff
@@ -5,11 +5,11 @@
 
 steps:
   - powershell: |
-      Install-Module -Name PSScriptAnalyzer -Repository PSGallery -Scope CurrentUser -AllowClobber -SkipPublisherCheck -Force -Verbose
+      Install-Module -Name PSScriptAnalyzer -Repository PSGallery -Scope CurrentUser -Force -Verbose
     displayName: 'Install PSScriptAnalyzer'
 
   - powershell: |
-       $results = Invoke-ScriptAnalyzer -Path ./ –Recurse
+       $results = try { Invoke-ScriptAnalyzer -Path ./ –Recurse -ErrorAction Stop } catch { $StackTrace; throw }
```

Comment:
I think that this change is that it will prevent us from getting our published NUnit results.  If that's the case, we wouldn't want to merge this in.  I can run this in CI a few more times though to see if we can catch the ScriptAnalyzer bug though.

Reply:
Why do you think it would prevent that? The point of re-throwing is to alert you that something bad happened during analysis so that you can then report the stack trace so that we can investigate further.
Right now, if that sporadic error happened in master, your build would be red as well, therefore I am not changing behaviour, I am only adding the additional information, which is the stack trace

Reply:
This could be my misunderstanding then of how `Invoke-ScriptAnalyzer` behaves with `-ErrorAction -Stop`.  I was under the impression that if it had any failures and `-ErrorAction Stop` was set, it would throw, causing any statements following it to not be executed. 

I just tried tried to verify this myself locally though, and it looks like my understanding there was wrong.  Can you just clarify under what scenarios `Invoke-ScriptAnalyzer -ErrorAction Stop` will result in an exception being thrown 

> Right now, if that sporadic error happened in master.

It has happened in PR CI builds as well.  There isn't anything special about the master builds vs the CI builds.

Reply:
Ok. Let me explain a few things first:
- `-ErrorAction` is a so called [common parameter](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_commonparameters?view=powershell-7) and not specific to PSScriptAnalyzer. Binary cmdlets like `Invoke-ScriptAnalyzer` or advanced script functions (i.e. function that are decorated with the `[CmdletBinding()]` attribute) automatically inherit those common parameters that the PowerShell engine handles/provides.
- There are 2 types of errors in PowerShell: terminating and non-terminating. The idea is that when an operation fails when processing pipeline input and one input fails, one usually does not want pipeline processing to stop because of that. This is important to know because:
- A `catch` block only catches terminating error and I know from experience that leaked exceptions from inside PSScriptAnalyzer are treated as non-terminating errors.
- Therefore we would not be able to catch that exception, which you sporadically observe just by doing `try { Invoke-ScriptAnalyzer } catch { 'hello' }`.
- You might be asking why the build is red then. This is because CI environment providers detect when a non-terminating errors occurred after a line has finished executing and fail the build as it is indication that something went wrong, which is right.
- Now, in order to catch and report the exception that happens in more detail, we need to catch it and by using `-ErrorAction Stop`, we can tell PowerShell to stop processing when any type of error occurs and treat them as terminating errors. This essentially allows us to get into the `catch` block

Therefore to conclude, this PR does not change the behaviour, it only logs out the stacktrace when an error occurs. Changing it to no re-throw to avoid the build going red would not be good because ScriptAnalyzer's analysis would be incomplete and we want to be alert of the error and report it the next time it happens. Therefore my suggestion is to just merge the PR and report the stack trace back to me the next time it happens so that we can look into fixing it in the next version

Reply:
Sounds good -- I appreciate the thorough response/help here.  Getting this merged-in now and will report back the stack trace should it happen again.